homeassistant/components/volvooncall.py

```diff
@@ -61,16 +46,16 @@
             vol.All(cv.time_period, vol.Clamp(min=MIN_UPDATE_INTERVAL))),
         vol.Optional(CONF_NAME, default={}): vol.Schema(
             {cv.slug: cv.string}),
-        vol.Optional(CONF_RESOURCES): vol.All(
-            cv.ensure_list, [vol.In(RESOURCES)]),
+        vol.Optional(CONF_RESOURCES): cv.ensure_list,
```

Comment:
Thanks for reviewing!
The idea here was that if someone added support for e.g. a switch volvo.sirens_and_rotating_blue_light in the external dependency, no update to the hass component (or documentation) would be neccessary, other than update the version number.

A list of resources names can of course be added (back) if this is considered bad design (there are some validation later anyway since different models supports different attributes).

Reply:
Generally we want to validate everything the user need to put in the config section. Sometimes that's not possible, but since we already have those values in the existing module, I suggest to keep this validation.

We would need to document it anyway in our docs, so also having it in the source doesn't add a lot of work overhead, and it will be beneficial for users.

Later, when we have an options flow for config entries, we could implement config entries for volvooncall and use the volvooncall library to validate the options.

Reply:
✓